docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -271,65 +370,27 @@ def validate_model_versions(self, versions):
                 return True
         return False
 
-    def _get_model_status(self, model_name):
-        url = 'http://localhost:{}/v1/models/{}'.format(TFS_REST_PORT, model_name)
-        try:
-            response = requests.get(url).json()
-            version_states = []
-            for version in response['model_version_status']:
-                version_states.append({
-                    'version': version['version'],
-                    'state': version['state']
-                })
-            return version_states
-        except Exception as e:  # pylint: disable=W0703
-            raise Exception(502, str(e))
-
-    def _check_all_versions_available(self, model_status):
-        if not model_status:
-            raise Exception(404, 'No model versions found')
-        for version in model_status:
-            if version['state'] != 'AVAILABLE':
-                return False
-        return True
-
-    def _check_all_versions_unloaded(self, model_status):
-        if not model_status:
-            raise Exception(404, 'No model versions found')
-        for version in model_status:
-            if version['state'] != 'END':
-                return False
-        return True
-
-    @contextmanager
-    def _timeout(self, seconds):
-        def _raise_timeout_error(signum, frame):
-            raise Exception(408, 'Timed our after {} seconds'.format(seconds))
 
-        try:
-            signal.signal(signal.SIGALRM, _raise_timeout_error)
-            signal.alarm(seconds)
-            yield
-        finally:
-            signal.alarm(0)
+class PingResource(object):
```

Comment:
will this execute in/is this still supporting python 2.7?

Reply:
No, TFS only supports py36, and py37 in recent versions.